### PR TITLE
Get wallet state from db

### PR DESCRIPTION
**Pull request by slundqui**
New feature to load wallet state, such that agent0 wallet states persist through reboots.

Major changes:
- Bots able to get current wallet positions (https://github.com/delvtech/elf-simulations/issues/850)
- System test for restarting bots to test getting current wallet positions
- Generalizing register username server to be db api server
- Adding in getting current wallet positions from db in db api server
- Building in db api server in test fixture

Minor changes
- Moving `FrozenClass`, `Quantity`, `TokenType`, `freezable`, `WalletDeltas`, `Long`, `Short` from elfpy to agent0
- Changing maturity time to be in `int` instead of `FixedPoint`
- Removing mint time in favor of maturity time for trades and wallets
- Moving wallet states: `agent0/base/agents -> agent0/base/state` and `agent0/hyperdrive/agents -> agent0/hyperdrive/state`

### CODE REVIEW

File: lib/agent0/agent0/base/agents/eth_wallet.py [hunk 1]
```diff
@@ -4,14 +4,17 @@
...
 
+from agent0.base import Quantity, TokenType
 from elfpy import check_non_zero
-from elfpy.types import Quantity, TokenType
-from elfpy.wallet.wallet_deltas import WalletDeltas
 from fixedpointmath import FixedPoint
 from hexbytes import HexBytes
 
+if TYPE_CHECKING:
+    # TODO need a base wallet delta
```
Comment (dpaiton) [inline lib/agent0/agent0/base/agents/eth_wallet.py:15]: I don't think we should leave this TODO -- this means there is a backwards dependency, where base is depending on hyperdrive. The base version is easy, it only has `balance`.

File: lib/agent0/agent0/hyperdrive/agents/__init__.py [hunk 2]
```diff
@@ -1,3 +1,3 @@
 """Account and wallet with Hyperdrive specific parts"""
 from .hyperdrive_account import HyperdriveAgent
-from .hyperdrive_wallet import HyperdriveWallet
+from .hyperdrive_wallet import HyperdriveWallet, Long, Short, WalletDeltas
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/agents/__init__.py:3]: per earlier comment: rename this to HyperdriveWalletDeltas & make a base class.

File: lib/agent0/agent0/hyperdrive/agents/hyperdrive_account.py [hunk 3]
```diff
@@ -92,7 +94,8 @@ def liquidation_trades(self) -> list[Trade[MarketAction]]:
                             action_type=MarketActionType.CLOSE_LONG,
                             trade_amount=long.balance,
                             wallet=self.wallet,  # type: ignore
-                            maturity_time=maturity_time,
+                            # TODO this should be in int
+                            maturity_time=FixedPoint(maturity_time),
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/agents/hyperdrive_account.py:98]: why can't it be an int now?

File: lib/agent0/agent0/hyperdrive/agents/hyperdrive_wallet.py [hunk 4]
```diff
@@ -71,7 +133,7 @@ def _update_longs(self, longs: Iterable[tuple[FixedPoint, Long]]) -> None:
             if maturity_time in self.longs and self.longs[maturity_time].balance < FixedPoint(0):
                 raise AssertionError(f"ERROR: Wallet balance should be >= 0, not {self.longs[maturity_time]}.")
 
-    def _update_shorts(self, shorts: Iterable[tuple[FixedPoint, Short]]) -> None:
+    def _update_shorts(self, shorts: Iterable[tuple[int, Short]]) -> None:
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/agents/hyperdrive_wallet.py:136]: fix docstrings

File: lib/agent0/agent0/hyperdrive/exec/execute_agent_trades.py [hunk 5]
```diff
@@ -294,7 +291,7 @@ async def async_match_contract_call_to_trade(
                     amount=trade_result.base_amount,
                     unit=TokenType.BASE,
                 ),
-                longs={trade.mint_time: Long(-trade_result.bond_amount)},
+                longs={trade.maturity_time: Long(-trade_result.bond_amount)},
```
Comment (dpaiton) [inline lib/agent0/agent0/hyperdrive/exec/execute_agent_trades.py:294]: I'm confused about whether this is an `int` or not -- there is a TODO elsewhere saying it should be an `int`, but it's a fixed point. In the hyperdrive wallet typedef it asks for `int`. A few lines back (264) in this file there is a new variable `maturity_time_seconds = int(trade.maturity_time)`, implying it is not an int.

File: lib/chainsync/chainsync/db/api/api_interface.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import pandas as pd
+import requests
+
+
+def register_username(api_uri: str, wallet_addrs: list[str], username: str) -> None:
+    """Registers the username with the flask server.
+
+    Arguments
+    ---------
+    register_uri: str
```
Comment (dpaiton) [inline lib/chainsync/chainsync/db/api/api_interface.py:15]: wrong var name

File: lib/chainsync/chainsync/test_fixtures/db_session.py [hunk 7]
```diff
@@ -71,11 +74,56 @@ def database_engine(psql_docker):
 
 @pytest.fixture(scope="function")
 def db_session(database_engine) -> Iterator[Session]:
-    """Initializes the in memory db session and creates the db schema"""
+    """Initializes the in memory db session and creates the db schema
```
Comment (dpaiton) [inline lib/chainsync/chainsync/test_fixtures/db_session.py:77]: add an `Arguments` docstring

**Review by dpaiton — CHANGES_REQUESTED** [review]
I left a few comments for small fixes. I think the only potentially large request is to resolve the TODO where one of the `maturity_time` types is still FixedPoint.

File: lib/agent0/agent0/hyperdrive/agents/hyperdrive_account.py (hunk 3 above)
Comment (slundqui) [reply lib/agent0/agent0/hyperdrive/agents/hyperdrive_account.py:98]: HyperdriveMarketAction is in elfpy, and we no touchy elfpy

**Pull request merged**